modules/packages/PrecisionSerializer.chpl

```diff
@@ -0,0 +1,575 @@
+      :arg size: The number of fields in the record.
+
+      :returns: A new :record:`AggregateSerializer`
+    */
+    proc startRecord(writer: fileWriter, name: string, size: int) throws {
+      writer.writeLiteral("(");
+      return new AggregateSerializer(writer, _ending=")");
+    }
+
+    /*
+      Returned by ``startClass`` or ``startRecord`` to provide the API for
+      serializing classes or records.
+
+      A ``class`` with integer fields 'x' and 'y' with values '0' and '5' would
+      be serialized as:
+
+      .. code-block:: text
+
+        {x = 0, y = 5}
+
+      A ``record`` with matching fields would be serialized in the same way, but
+      would use ``(`` and ``)`` instead of ``{`` and ``}``.
+    */
+    record AggregateSerializer {
+      @chpldoc.nodoc
+      var writer;
+      @chpldoc.nodoc
+      var _parent = false;
+      @chpldoc.nodoc
+      var _first : bool = true;
+      @chpldoc.nodoc
+      const _ending : string;
+
+      // pointer to child's 'first' field so that we can communicate back if/when
+      // a field has already been written.
+      @chpldoc.nodoc
+      var _firstPtr : c_ptr(bool) = nil;
+
+      /*
+        Serialize ``field`` named ``name``.
+
+        Serializes fields in the form '<name> = <field>'. Adds a comma before the
+        name if this is not the first field.
+      */
+      proc ref writeField(name: string, const field: ?) throws {
+        if !_first then writer.writeLiteral(", ");
+        else _first = false;
+
+        writer.writeLiteral(name);
+        writer.writeLiteral(" = ");
+        writer.write(field);
+      }
+
+      /*
+        Start serializing a nested class inside the current class. In this format
+        inheritance is not represented and parent fields are printed before child
+        fields. For example, the following classes with values
+        ``x=5`` and ``y=2.0``:
+
+        .. code-block:: chapel
+
+          class Parent {
+            var x : int;
+          }
+
+          class Child: Parent {
+            var y : real;
+          }
+
+        would be serialized as:
+
+        .. code-block:: text
+
+          {x = 5, y = 2.000000000000000}
+
+        :arg writer: The :record:`~IO.fileWriter` to be used when serializing. Must match
+                    the writer used to create current AggregateSerializer.
+        :arg name: The name of the class type.
+        :arg size: The number of fields in the class.
+
+        :returns: A new :record:`~PrecisionSerializer.precisionSerializer.AggregateSerializer`
+      */
+      proc ref startClass(writer: fileWriter, name: string, size: int) throws {
+        // Note: 'size' of parent might be zero, but 'size' of grandparent might
+        // be non-zero.
+        return new AggregateSerializer(writer, _parent=true,
+                                      _firstPtr=c_addrOf(_first));
+      }
+
+      /*
+        Ends serialization of the current class by writing the character ``}``
```

Comment:
The indentation of this text in your generated docs is weird (but that's not your fault).  Would you be up for filing a bug against chpldoc on it?

Reply:
issue here: https://github.com/chapel-lang/chapel/issues/25540